app/src/main/java/org/mozilla/tv/firefox/navigationoverlay/BrowserNavigationOverlay.kt

```diff
@@ -225,19 +225,26 @@ class BrowserNavigationOverlay @JvmOverloads constructor(
     fun showMegaTileError() {
         pocketVideosContainer.visibility = View.GONE
         pocketErrorContainer.visibility = View.VISIBLE
+
         pocketMegaTileLoadError.text = resources.getString(R.string.pocket_video_feed_failed_to_load,
                 resources.getString(R.string.pocket_brand_name))
         megaTileTryAgainButton.contentDescription = resources.getString(R.string.pocket_video_feed_failed_to_load,
                 resources.getString(R.string.pocket_brand_name)) + " " + resources.getString(R.string.pocket_video_feed_reload_button)
 
         megaTileTryAgainButton.setOnClickListener { _ ->
+            hideMegaTileError()
             pocketViewModel.update()
             initMegaTile()
             updateOverlayForCurrentState()
-            pocketVideoMegaTileView.requestFocus()
         }
     }
 
+    private fun hideMegaTileError() {
+        pocketVideosContainer.visibility = View.VISIBLE
+        pocketErrorContainer.visibility = View.GONE
+        pocketVideoMegaTileView.requestFocus()
```

Comment:
This isn't an ideal solution, because we're duplicating visibility state between the VM and the fragment.  With this change, state in these two places could go out of sync, leading to difficult to find bugs.  Also this puts very similar code into two different places, complicating the process of making further changes.

A better solution would be to move this into the `pocketViewModel.state` observer in  `BrowserNavigationOverlay#observeForMegaTile`, so that our displayed UI would be modeled directly by our VM state.